### PR TITLE
Support multiple dataframe input to CE construct

### PR DESCRIPTION
**Pull request by adamnsch**
Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

**Comment by Mats-SX** [comment]
The test failure is an unrelated test that is also failing on main, which @DarthMax is working to resolve.

File: graphdatascience/query_runner/cypher_graph_constructor.py [hunk 1]
```diff
@@ -46,23 +77,25 @@ def run(self, node_dfs: List[DataFrame], relationship_dfs: List[DataFrame]) -> N
...
+            ).run(node_dfs, relationship_dfs)
         else:
             assert not self._undirected_relationship_types, "This should have been raised earlier."
 
+            if len(node_dfs) > 1:
+                raise ValueError("The GDS Community edition graph construction supports only a single node dataframe")
+
+            if len(relationship_dfs) > 1:
+                raise ValueError(
+                    "The GDS Community edition graph construction supports at most one relationship dataframe"
```
Comment (Mats-SX) [inline graphdatascience/query_runner/cypher_graph_constructor.py:93]: We should indicate in the error message that this limitation is tied to GDS versions prior to 2.3

File: graphdatascience/query_runner/cypher_graph_constructor.py [hunk 2]
```diff
@@ -99,45 +132,55 @@ def __init__(
             self._graph_name = graph_name
             self._undirected_relationship_types = undirected_relationship_types
 
-        def run(self, node_df: DataFrame, relationship_df: DataFrame) -> None:
-            graph_schema = self.schema(node_df, relationship_df)
+        def run(self, node_dfs: List[DataFrame], relationship_dfs: List[DataFrame]) -> None:
+            graph_schema, dummy_values = self.schema(node_dfs, relationship_dfs)
```
Comment (Mats-SX) [inline graphdatascience/query_runner/cypher_graph_constructor.py:136]: I think we do not use `dummy_values` anywhere, do we?

File: graphdatascience/tests/unit/test_graph_construct.py [hunk 3]
```diff
@@ -80,28 +131,51 @@ def test_graph_aggregation_based_alpha_construct_without_arrow(
...
+            None,
+            False,
+            1,
+            ["B"],
+            True,
+            {"propF": 42.42, "propI": 42, "propList": [1, 2, 3]},
+            True,
+            -1,
+            False,
+        ],
```
Comment (Mats-SX) [inline graphdatascience/tests/unit/test_graph_construct.py:176]: we should remove some linebreaks here to make this readable

**Comment by Mats-SX** [comment]
Rebased to get the fix for the failing test

**Review by adamnsch — COMMENTED** [review]
Apparently I cannot Approve my own PR, so I'll just say it: Approved! :heavy_check_mark:

**Pull request merged**